src/libraries/System.Private.CoreLib/src/System/CharEnumerator.cs

```diff
@@ -18,6 +18,11 @@ public sealed class CharEnumerator : IEnumerator, IEnumerator<char>, IDisposable
 
         public bool MoveNext()
         {
+            if (_str == null)
```

Comment:
A null check would have performance impact on usual case.
The original issue suggests changing null to empty string in `Dispose`.